regolith/tools.py

```diff
@@ -2110,6 +2122,143 @@ def google_cal_auth_flow():
+                            return False
+                    else:
+                        print(message_url_not_defined)
+                        return False
+                else:
+                    print()
+                    return False
+            else:
+                print(setup_message)
+                return False
+        else:
+            print(setup_message)
+            return False
+    else:
+        print(setup_message)
+        return False
+
+
+def token_info_complete(token_info, rc):
+    """Checks if API authentication token is defined and valid in rc
+
+    Parameters:
+        token_info - string
+            the personal access token defined in rc (e.g. 'priv_token' or 'gitlab_private_token')
+        rc - run control object
+
+    Returns:
+        The token if the token exists and False if not
+    """
+    message_token_not_defined = ("WARNING: The authentication token may not be defined. If you would like regolith to "
+                                 "automatically create a repository in GitHub/GitLab, please add your private authentication "
+                                 "token in user.json. See regolith documentation for details.")
+    if token_info in rc:
+        if rc.__getattr__(token_info):
+            return rc.__getattr__(token_info)
+        else:
+            print(message_token_not_defined)
+            return False
+    else:
+        print(message_token_not_defined)
+        return False
+
+
+def create_repo(name, target_repo, token_info, rc):
+    """ Creates a repo if repo information and token is defined in rc
+
+    Parameters:
+        name - string
+            the name of the repo being created (e.g. 'test repo')
+        target_repo - string
+            the key in rc for target repo information (e.g. 'repo1')
+        token_info - string
+            the key in rc for token (e.g. 'priv_token')
+        rc - run control object
+
+    Returns:
+        Success message (repo target_repo has been created in talks) if repo is successfully created in target_repo
+        Warning/setup messages if unsuccessful (or if repo info or token are not valid)
+    """
+    repo_info_exists = repo_info_complete(target_repo, rc)
+    token_exists = token_info_complete(token_info, rc)
+    if repo_info_exists and token_exists:
+        try:
+            response = requests.post(rc.repos[target_repo]['url'], params=rc.repos[target_repo]['params'],
+                                     headers={'PRIVATE-TOKEN': '{}'.format(token_exists)})
+            response.raise_for_status()
+            return f"repo {name} has been created in talks"
+        except requests.exceptions.HTTPError:
+            raise HTTPError(f"WARNING: Unsuccessful attempt at making a GitHub/GitLab etc., repository "
+                            f"due to an issue with the API call (status code: {response.status_code}). "
+                            f"Check that your private token and repository information are valid in "
```

Comment:
this is confusing because private token should NEVER be in regolithrc.json but the wording here makes it sound to a user like it might be.....please update the wording so this is handled explicitly.  Best may be to link to a place where the documentation is rather than making a really long message here.